connmgr.go

```diff
@@ -155,7 +156,15 @@ func (cm *BasicConnMgr) getConnsToClose(ctx context.Context) []inet.Conn {
 
 	// Sort peers according to their value.
 	sort.Slice(candidates, func(i, j int) bool {
-		return candidates[i].value < candidates[j].value
+		left, right := candidates[i], candidates[j]
+		// temporary peers are preferred for pruning.
```

Comment:
Why are they preferred though? I think we may as well just treat them as regular tags.

Reply:
Early tags are allowed to exist until the grace period expires. This is defensive programming: we need them to be temporary to avoid memory leaks. Otherwise, in the case no `Connected` callback is ever fired for that peer, they could linger forever (if their value is high enough). An example is a protocol that speculatively tags peers before establishing a connection; if the connection fails and the protocol doesn't untag it, the connection manager will catch it with this logic. If not, it'll stockpile and intermix with good tags.

Reply:
Ok, fair enough.

Reply:
```go
// of one is temp and the other is not
if left.temp != right.temp {
  // the temp one is "less"
  return left.temp
}
// otherwise, compare by value.
return left.value < right.value
```

Reply:
duh